onboarding/src/main/kotlin/io/homeassistant/companion/android/onboarding/nameyourdevice/NameYourDeviceScreen.kt

```diff
@@ -0,0 +1,211 @@
+    deviceName: String,
+    onDeviceNameChange: (String) -> Unit,
+    saveClickable: Boolean,
+    deviceNameEditable: Boolean,
+    onSaveClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .padding(horizontal = HASpacing.M)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(HASpacing.XL),
+    ) {
+        Image(
+            modifier = Modifier.padding(top = HASpacing.XL),
+            // Use painterResource instead of vector resource for API < 24 since it has gradients
+            painter = painterResource(R.drawable.ic_name_tag),
+            contentDescription = null,
+        )
+        Text(
+            text = stringResource(R.string.name_your_device_title),
+            style = HATextStyle.Headline,
+        )
+        Text(
+            text = stringResource(R.string.name_your_device_content),
+            style = HATextStyle.Body,
+        )
+
+        DeviceNameTextField(
+            deviceName = deviceName,
+            onDeviceNameChange = onDeviceNameChange,
+            saveClickable = saveClickable,
+            onSaveClick = onSaveClick,
+            deviceNameEditable = deviceNameEditable,
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        HAAccentButton(
+            text = stringResource(R.string.name_your_device_save),
+            onClick = onSaveClick,
+            enabled = saveClickable,
+            modifier = Modifier
+                .padding(bottom = HASpacing.XL),
+        )
+    }
+}
+
+@Composable
+private fun DeviceNameTextField(
+    deviceName: String,
+    onDeviceNameChange: (String) -> Unit,
+    saveClickable: Boolean,
+    deviceNameEditable: Boolean,
+    onSaveClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val focusRequester = remember { FocusRequester() }
+
+    HATextField(
+        value = deviceName,
+        onValueChange = onDeviceNameChange,
+        trailingIcon = {
+            if (deviceName.isNotEmpty()) {
+                IconButton(onClick = { onDeviceNameChange("") }, enabled = deviceNameEditable) {
+                    Icon(
+                        imageVector = Icons.Default.Close,
+                        contentDescription = stringResource(R.string.name_your_device_clear_name),
+                    )
+                }
+            }
+        },
+        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+        keyboardActions = KeyboardActions(
+            onDone = {
+                focusRequester.freeFocus()
+                if (saveClickable) {
+                    onSaveClick()
+                }
+                // This is going to hide the keyboard and clear focus on the text field
+                defaultKeyboardAction(ImeAction.Done)
+            },
+        ),
+        enabled = deviceNameEditable,
```

Comment:
This does not disable the cross to clear the field input. I'm not sure whether that should be handled by this option, or in the trailingIcon compose function.